### PR TITLE
feat: Support to verify authorized party (azp) session token claim

### PR DESCRIPTION
**Pull request by chanioxaris**
## Type of change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 🌟 New feature (non-breaking change which adds functionality)
- [ ] 🔨 Breaking change (fix or feature that would cause existing functionality)
- [ ] 📖 Docs change / refactoring / dependency upgrade to change)

## Description

Added support to verify the authorized party (azp) claim, if it's included in the session token claims within the VerifyToken method and v2 middlewares

### CODE REVIEW

**Review by gkats — APPROVED** [review]

**Comment by SokratisVidros** [comment]
@chanioxaris Is this added to node SDK as well? The only one left should be Ruby, right?

**Comment by chanioxaris** [comment]
> @chanioxaris Is this added to node SDK as well? The only one left should be Ruby, right?

Not yet, it's under implementation

File: clerk/tokens.go [hunk 1]
```diff
@@ -46,8 +47,18 @@ func (c *client) DecodeToken(token string) (*TokenClaims, error) {
 	return &TokenClaims{Claims: standardClaims, Extra: extraClaims}, nil
 }
 
+type verifyTokenOptions struct {
+	authorizedParties []string
```
Comment (agis) [inline clerk/tokens.go:51]: Can we make this a map so that we avoid the linear search?

Comment (chanioxaris) [reply clerk/tokens.go]: Nice catch, changed in [10dc7fa](https://github.com/clerkinc/clerk-sdk-go/pull/43/commits/10dc7fa305c1b2fffa2993a37f93b1f92461ece1)

**Pull request merged**